AK/Try.h

```diff
@@ -15,22 +15,30 @@
 //       on statement expressions [1]. This is known to be implemented
 //       by at least clang and gcc.
 //       [1] https://gcc.gnu.org/onlinedocs/gcc/Statement-Exprs.html
+//
+//       If the static_assert below is triggered, it means you tried to return a reference
+//       from a fallible expression. This will not do what you want; the statement expression
+//       will create a copy regardless, so it is explicitly disallowed.
 
-#define TRY(expression)                                   \
-    ({                                                    \
-        /* Ignore -Wshadow to allow nesting the macro. */ \
-        AK_IGNORE_DIAGNOSTIC("-Wshadow",                  \
-            auto _temporary_result = (expression));       \
-        if (_temporary_result.is_error()) [[unlikely]]    \
-            return _temporary_result.release_error();     \
-        _temporary_result.release_value();                \
+#define TRY(expression)                                                                \
+    ({                                                                                 \
+        /* Ignore -Wshadow to allow nesting the macro. */                              \
+        AK_IGNORE_DIAGNOSTIC("-Wshadow",                                               \
+            auto _temporary_result = (expression));                                    \
+        static_assert(!IsLvalueReference<decltype(_temporary_result.release_value())>, \
+            "Do not return a reference from a fallible expression");                   \
+        if (_temporary_result.is_error()) [[unlikely]]                                 \
+            return _temporary_result.release_error();                                  \
+        _temporary_result.release_value();                                             \
     })
 
-#define MUST(expression)                                  \
-    ({                                                    \
-        /* Ignore -Wshadow to allow nesting the macro. */ \
-        AK_IGNORE_DIAGNOSTIC("-Wshadow",                  \
-            auto _temporary_result = (expression));       \
-        VERIFY(!_temporary_result.is_error());            \
-        _temporary_result.release_value();                \
+#define MUST(expression)                                                               \
+    ({                                                                                 \
+        /* Ignore -Wshadow to allow nesting the macro. */                              \
+        AK_IGNORE_DIAGNOSTIC("-Wshadow",                                               \
+            auto _temporary_result = (expression));                                    \
+        static_assert(!IsLvalueReference<decltype(_temporary_result.release_value())>, \
```

Comment:
Can we do the same for the OOM-error and and possibly other TRY macro variants please?